File: app/src/main/java/co/tpcreative/supersafe/common/controller/SingletonManager.java
package co.tpcreative.supersafe.common.controller;
public class SingletonManager {
    private static SingletonManager instance ;
    public boolean isAnimation() {
        return isAnimation;
    }
    public void setAnimation(boolean isAnimation) {
        this.isAnimation = isAnimation;
    }
    private boolean isAnimation;
    public boolean isVisitLockScreen() {
        return isVisitLockScreen;
    }

    public void setVisitLockScreen(boolean visitLockScreen) {
        isVisitLockScreen = visitLockScreen;
    }

    public boolean isVisitFakePin() {
        return isVisitFakePin;
    }

    public void setVisitFakePin(boolean visitFakePin) {
        isVisitFakePin = visitFakePin;
    }

    public boolean isReloadMainTab() {
        return isReloadMainTab;
    }

    public void setReloadMainTab(boolean reloadMainTab) {
        isReloadMainTab = reloadMainTab;
    }

    private boolean isVisitLockScreen;
    private boolean isVisitFakePin;
    private boolean isReloadMainTab;

    public static SingletonManager getInstance(){
        if (instance==null){
            synchronized (SingletonManager.class){
                if (instance==null){
                    instance = new SingletonManager();
                }
            }
        }
        return instance;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/common/controller/SingletonManager.kt
package co.tpcreative.supersafe.common.controller

class SingletonManager {
    fun isVisitLockScreen(): Boolean {
        return isVisitLockScreen
    }

    fun setVisitLockScreen(visitLockScreen: Boolean) {
        isVisitLockScreen = visitLockScreen
    }

    fun isVisitFakePin(): Boolean {
        return isVisitFakePin
    }

    fun setVisitFakePin(visitFakePin: Boolean) {
        isVisitFakePin = visitFakePin
    }

    fun isReloadMainTab(): Boolean {
        return isReloadMainTab
    }

    fun setReloadMainTab(reloadMainTab: Boolean) {
        isReloadMainTab = reloadMainTab
    }

    private var isVisitLockScreen = false
    private var isVisitFakePin = false
    private var isReloadMainTab = false

    companion object {
        private var instance: SingletonManager? = null
        fun getInstance(): SingletonManager {
            if (instance == null) {
                synchronized(SingletonManager::class.java) {
                    if (instance == null) {
                        instance = SingletonManager()
                    }
                }
            }
            return instance!!
        }
    }
}